### ISSUE TITLE
Surface Pro 4 Won't boot after installing Surface kernel

### ISSUE THREAD

**Issue by N0tallion6** [linux-surface/linux-surface#1134]
Recently I decided to grab a 2nd hand surface pro 4 and put Arch Linux on it.
I've followed the instructions up until the point where you install `surface-linux`, `surface-linux-headers` and `iptsd`. Also installed the `linux-firmware-marvell` package for good measure.
I haven't set up secure boot yet as I wanted to make sure it was working properly after the previous steps first.

For reasons I'm not too sure about, after doing this and rebooting, I get the following error when trying to boot the machine:
`warning: /lib/modules/6.2.13-arch1-2-surface/modules.devname not found - ignoring`
`Starting systemd-udevd version 253.4-1-arch`

After that it tries to mount my root partition, (checked the UUID against my fstab) fails to do so and drops me into the emergency shell. From there it won't take any input (type cover or external USB keyboard) leaving me with no choice but to reboot by holding down the power button until it shuts off.

The issue appears to be the `linux-surface` package, as uninstalling it through the LiveUSB & Arch-chroot appears to rectify the issue.

Have I missed anything?

If it helps at all, I'm using rEFInd as my bootloader, though I don't see how that would affect it, 
That said, it doesn't appear to have any options to boot with the stock Arch kernel instead of the surface kernel.

Thanks in advance

**Comment by qzed** [comment]
That sounds like something went wrong during the installation of the `linux-surface` package. Can you try reinstalling it and see if that helps?

Also: You shouldn't need to boot via a live iso, the linux-surface kernel is installed alongside the standard Arch kernel, so you should be able to select a different one via your bootloader.

**Comment by N0tallion6** [comment]
> That sounds like something went wrong during the installation of the `linux-surface` package. Can you try reinstalling it and see if that helps?

Just tried that, as well as re-installing my bootloader. Instead of telling me that it's waiting 10 seconds for the hard drive and then expiring, it just tells me `ERROR: device '' not found. skipping fsck.` and `mount: /new_root: no file system type specified` then drops me into another emergency shell.

The good news is that now I have additional options in rEFInd that allows me to boot with the stock kernel now, though it seems to be a little inconsistent as sometimes it will boot fine and other times give me the same error as above.

**Comment by qzed** [comment]
Okay, so that now looks like you may have specified the wrong `root` drive in the kernel boot parameters. In particular, I think the `device ''` might indicate that it may be unspecified. Did you by any chance run `refind-install` inside a chroot environment, e.g. from the live-disk? I believe that might cause something like that. See this warning from the [Arch wiki page](https://wiki.archlinux.org/title/REFInd) on rEFInd:

> Warning: When refind-install is run in chroot (e.g. in live system when installing Arch Linux) /boot/refind_linux.conf is populated with kernel options from the live system not the one on which it is installed. Edit /boot/refind_linux.conf and make sure the [kernel parameters](https://wiki.archlinux.org/title/Kernel_parameters) in it are correct for your system, otherwise you could get a kernel panic on your next boot. See [#refind_linux.conf](https://wiki.archlinux.org/title/REFInd#refind_linux.conf) for an example file.

**Comment by N0tallion6** [comment]
Ah, that might be it! I believe I did run `refind-install` in arch-chroot from my liveUSB when I initially set up Arch. I must've missed something.
I inspected my /boot/refind_linux.conf and this is what it looks like:

`"Boot with standard options" "archisobasedir=arch archisodevice=UUID=2023-05-03-17-21-52-00"`
`"Boot to single-user mode"     "archisobasedir=arch archisodevice=UUID=2023-05-03-17-21-52-00 single"`
`"Boot with minimal options"   "rw root=/dev/nvme0n1p6"`

(nvme0n1p6 is my root partition, the efi is nvme0n1p5)

I'm still new to refind, but that UUID seems to be a timestamp - doesn't look right to me.
I guess I need to replace that with something more appropriate? I tried replacing the parts in quotes with the UUID of my root partition, similar to what was shown in the example but no dice. 
Not sure if I'm doing it correctly, it's getting late where I am, so I'll re-read the manual with a fresh pair of eyes tomorrow, see if I can make sense of it...

Thanks for pointing out the Arch-Chroot issue - don't know how I missed that on the Wiki!

**Comment by N0tallion6** [comment]
Managed to sort it! I played around with my `refind_linux.conf` and managed to get further with the boot process and I've started seeing the systemd logs on my screen telling me that everything is starting/has started successfully.

`"boot with defaults"    "root=UUID=<uuid of my root drive> nomodeset rw"`

I ran into a further issue where it gave the below errors, but was able to fix that by removing `nomodeset` from my `refind_linux.config`
`Bluetooth: hci0: unexpected event for opcode 0x0000` (This is listed twice for some reason)
`hdaudio hdaudioC0D2: Unable to configure, disabling`

Thanks very much for your help @qzed!
